jsonnet/utils/configure-authentication-for-monitors.libsonnet

```diff
@@ -14,6 +14,7 @@
                           if !(std.objectHas(o.metadata.labels, 'app.kubernetes.io/name') && o.metadata.labels['app.kubernetes.io/name'] == 'kubelet') then
                             {
                               caFile: '/etc/prometheus/configmaps/serving-certs-ca-bundle/service-ca.crt',
+                              serverName: std.format('%s.%s.svc', [if o.metadata.name != 'thanos-sidecar' then o.metadata.name else 'prometheus-k8-' + o.metadata.name, o.metadata.namespace]),
```

Comment:
yes, should be `prometheus-k8s`, not `prometheus-k8`, tested with PR
```
$ oc -n openshift-monitoring get servicemonitor thanos-sidecar -oyaml
..
      serverName: prometheus-k8-thanos-sidecar.openshift-monitoring.svc
```
this would cause `thanos-sidecar` target down
```
        "labels": {
          "container": "kube-rbac-proxy-thanos",
          "endpoint": "thanos-proxy",
          "instance": "10.128.2.16:10902",
          "job": "prometheus-k8s-thanos-sidecar",
          "namespace": "openshift-monitoring",
          "pod": "prometheus-k8s-0",
          "service": "prometheus-k8s-thanos-sidecar"
        },
        "scrapePool": "serviceMonitor/openshift-monitoring/thanos-sidecar/0",
        "scrapeUrl": "https://10.128.2.16:10902/metrics",
        "globalUrl": "https://10.128.2.16:10902/metrics",
        "lastError": "Get \"https://10.128.2.16:10902/metrics\": x509: certificate is valid for *.prometheus-k8s-thanos-sidecar.openshift-monitoring.svc, *.prometheus-k8s-thanos-sidecar.openshift-monitoring.svc.cluster.local, prometheus-k8s-thanos-sidecar.openshift-monitoring.svc, prometheus-k8s-thanos-sidecar.openshift-monitoring.svc.cluster.local, not prometheus-k8-thanos-sidecar.openshift-monitoring.svc",
        "lastScrape": "2023-02-01T09:48:18.63142094Z",
        "lastScrapeDuration": 0.002260885,
        "health": "down",
        "scrapeInterval": "30s",
        "scrapeTimeout": "10s"
      },
```

Reply:
Missed that typo, should be fixed